cmd/wio/commands/run/dependencies/dep.go

```diff
@@ -0,0 +1,145 @@
+package dependencies
+
+import (
+    "fmt"
+    "path/filepath"
+    "strings"
+    "wio/cmd/wio/commands/run/cmake"
+    "wio/cmd/wio/constants"
+    "wio/cmd/wio/errors"
+    "wio/cmd/wio/log"
+    "wio/cmd/wio/toolchain/npm/resolve"
+    "wio/cmd/wio/types"
+    "wio/cmd/wio/utils"
+    "wio/cmd/wio/utils/io"
+    "wio/cmd/wio/utils/template"
+)
+
+const (
+    MainTarget = "${TARGET_NAME}"
+)
+
+var libraryStrings = map[string]map[bool]string{
+    "avr":    {false: cmake.AvrLibrary, true: cmake.AvrHeader},
+    "native": {false: cmake.DesktopLibrary, true: cmake.DesktopHeader},
+}
+
+// This creates CMake dependency string using build targets that will be used to link dependencies
+func GenerateCMakeDependencies(cmakePath string, platform string, targets *TargetSet) error {
+    cmakeStrings := make([]string, 0, 256)
+
+    for target := range targets.TargetIterator() {
+        finalString := libraryStrings[platform][target.HeaderOnly]
+
+        if target.HeaderOnly && target.FlagsVisibility != "INTERFACE" {
+            return errors.Stringf("%s@%s dependency is header only so flags visibility can only be INTERFACE",
+                target.Name, target.Version)
+        } else if target.HeaderOnly && target.DefinitionsVisibility != "INTERFACE" {
+            return errors.Stringf("%s@%s dependency is header only so definitions visibility can only be INTERFACE",
+                target.Name, target.Version)
+        } else if strings.Trim(target.FlagsVisibility, " ") == "" {
+            log.Warnln(fmt.Sprintf("flags visibility not specified for %s@%s dependency. Defaulting to PRIVATE",
```

Comment:
`Warnln` can accept parameters like `log.Warnln(flags visibility not specified for %s@%s", target.Name, target.Version)`

Reply:
I will make that change